src/h5web/visualizations/heatmap/ColorMapSelector.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { CSSProperties } from 'react';
+import Select from 'react-select';
+import { generateCSSLinearGradient } from './utils';
+import { ColorMap, COLOR_MAPPINGS } from './models';
+
+interface Props {
+  className: string;
+  currentColorMap: ColorMap;
+  changeColorMap: React.Dispatch<React.SetStateAction<ColorMap>>;
+}
+
+// react-select expects to work with objects of type {label: string, value: string}
+// So I use the color map name as both label and value
+const colorMapOptions = Object.keys(COLOR_MAPPINGS).map(label => {
+  return { label };
+});
+
+function ColorMapSelector(props: Props): JSX.Element {
+  const { className, currentColorMap, changeColorMap } = props;
+
+  const gradientStyles = {
+    option: (
+      styles: CSSProperties,
+      { data }: { data: { label: ColorMap } }
+    ) => {
+      return {
+        ...styles,
+        backgroundImage: generateCSSLinearGradient(
+          COLOR_MAPPINGS[data.label],
+          'right'
+        ),
+        marginTop: '1px',
+        marginBottom: '1px',
+      };
+    },
+  };
+  return (
+    <div className={className}>
+      <Select
+        defaultValue={{ label: currentColorMap }}
+        options={colorMapOptions}
+        onChange={selection => {
+          const colorOption = selection as { label: ColorMap };
+          changeColorMap(colorOption.label);
+        }}
+        menuPlacement="top"
+        styles={gradientStyles}
+      />
```

Comment:
FYI, when you'll refactor the design, it may be easier [to replace](https://react-select.com/components#replacing-components) react-select's default `Option` component instead of overriding the styles. Perhaps you were already planning to.

Reply:
Thanks :+1: !